Assignment 3

Carvas91 opened the pull request:
## What changes are you trying to make? (e.g. Adding or removing code, refactoring existing code, adding reports)
I added the code for the K mean clustering and for the confidence interval calculations.

## What did you learn from the changes you have made?
I learned what bootstrap sample distributions are for. I found this concept very interesting. Honestly, I've seen this concept before but now I was able to understand it better.

## Was there another approach you were thinking about making? If so, what approach(es) were you thinking of?
Not that I can think of.

## Were there any challenges? If so, what issue(s) did you face? How did you overcome it?
At first it was understanding why we calculate the bootstrapped mean and the confidence intervals. After paying full attention to the class, asking questions and researching, I finally understand.

## How were these changes tested?
I executed the cells in the notebook and checked if the outcome looked correct.

## A reference to a related issue in your repository (if applicable)
No issues.

## Checklist
- ✅ I can confirm that my changes are working as intended

Carvas91 committed:
repo synced

Carvas91 committed:
question 3 answered

Carvas91 committed:
question 4 answered

Carvas91 committed:
assignment completed

Dmytro-Bonislavskyi reviewed (changes requested):
Hi @Carvas91 

The idea was to bootstrap a number of random samples from iris_df['petal width (cm)'].
However, your approach of bootstrapping from one random sample taken from the main sample is also nice:

```
for i in range(20_000):
    sample = one_sample.sample(frac=1, replace=True)
```
So, considering the ambiguity in the task, I can except it :)

There’s still a very small issue that still needs to be addressed:

While calculating the mean of Petal Width, you are using `iris_df['petal width (cm)'] `- the sample provided before bootstrapping.
`mean_petal_width = iris_df['petal width (cm)'].mean()`

However, the task required calculating the mean of the bootstrapped samples:
bootstrap_means.mean()

Please make this correction and push the new commit. Then we’re done with LCR :)

Carvas91 committed:
I updated the code to show the mean of the bootstraped samples print(f'The mean of the bootstrapped samples is {bootstrap_means["mean_petal_width"].mean():.3f}')

Carvas91 commented:
Dear @Dmytro-Bonislavskyi Thank you for your feedback! I've just updated the code to display the mean of the bootstrapped samples, which falls within the confidence interval range. I also modified the code to generate only 10,000 bootstrap samples (instead of 20k) and updated the print statements to show three decimal places.

Dmytro-Bonislavskyi reviewed (approved):
Well done!
